src/tsolvers/liasolver/LIACutSolver.cpp

```diff
@@ -0,0 +1,505 @@
+                v_found = var_nonbasic;
+                eqBasisSimplex.pivot(var, var_nonbasic);
+            }
+        }
+    }
+}
+
+void LIACutSolver::initialize(Simplex& eqBasisSimplex, LRAModel& eqBasisModel) {
+    //for all vars be it basic or non-basic
+    const vec<LVRef> & allVars = eqBasisModel.getAllVars();
+    for (int i = 0; i < allVars.size(); i++) {
+        LVRef var = allVars[i];
+        if (eqBasisModel.read(var) > eqBasisModel.Lb(var)) {
+            eqBasisModel.clearLBound(var);
+        }
+        if (eqBasisModel.read(var) < eqBasisModel.Ub(var)) {
+            eqBasisModel.clearUBound(var);
+        }
+        if ((eqBasisModel.read(var).hasDelta())) {
+            eqBasisModel.clearUBound(var);
+            eqBasisModel.clearLBound(var);
+        }
+    }
+
+    for (auto var : eqBasisSimplex.getBasicVars()) {
+        //for basicvar in basic
+        if (eqBasisModel.Lb(var) == eqBasisModel.Ub(var))
+            pivotOnSmallestConsistentInPoly(var, eqBasisSimplex);
+    }
+
+    for (int i = 0; i < allVars.size(); i++) {
+        LVRef var = allVars[i];
+        if (eqBasisModel.Lb(var) >= eqBasisModel.Ub(var))
+            continue;
+
+        //for x_k in basic union nonbasic
+        if (!eqBasisModel.Lb(var).isMinusInf()){//this checks if lower bound of var is minus infinite
+            //l_k := l_k +sigma
+            eqBasisModel.tightenLBound(var);
+            //assert(eqBasisModel.Lb(var).getType() == bound_l);
+            assert(eqBasisModel.readLBound(var).getType() == bound_l);
+        }
+        if (!eqBasisModel.Ub(var).isPlusInf()){
+            //u_k := u_k - sigma
+            eqBasisModel.tightenUBound(var);
+            //assert(eqBasisModel.Ub(var).getType() == bound_u);
+            assert(eqBasisModel.readUBound(var).getType() == bound_l);
+        }
+
+        // What's the point of these?
+        if (!eqBasisModel.Lb(var).isMinusInf() && eqBasisSimplex.isNonBasic(var)){
+            eqBasisSimplex.changeValueBy(var, eqBasisModel.Lb(var));
+        }
+        if (!eqBasisModel.Ub(var).isPlusInf() && eqBasisSimplex.isNonBasic(var)){
+            //updateValues(x_k, u_k)
+            eqBasisSimplex.changeValueBy(var, eqBasisModel.Ub(var));
+        }
+    }
+}
+
+void LIACutSolver::fixEqualities(LVRef basicVar, Simplex& s, LRAModel& m){
+    //input is basic var that explains the conflict form Simplex when it returns UNSAT
+    //turns tightly bounded vars into explicit equalities
+    assert(s.isBasic(basicVar));
+    //for var that belongs to Nonbasic vars
+    for (auto const &term : s.getRowPoly(basicVar)) {
+        auto var = term.var;
+        assert(s.isNonBasic(var));
+        assert(var != basicVar);
+        auto const &coeff = term.coeff;
+        const bool is_coeff_pos = coeff > 0;
+        if (m.Lb(var) < m.Ub(var)) {
+            if (is_coeff_pos && m.read(basicVar) < m.Lb(basicVar)) {
+                m.relaxUBound(var);
+                m.setLBoundToUBound(var);
```

Comment:
Also these two methods: relaxUBound and setLBoundToUBound seem new and not present in current code.

Reply:
See above.